core/src/com/biegan/game/Controller/GameController.java

```diff
@@ -0,0 +1,283 @@
+package com.biegan.game.Controller;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.audio.Music;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Timer;
+import com.biegan.game.BieganBrickRaceGame;
+import com.biegan.game.Model.EnemyCarManager;
+import com.biegan.game.Model.GameModel;
+import com.biegan.game.View.Hud;
+import com.biegan.game.View.screens.GameOverScreen;
+import com.biegan.game.View.screens.GameScreen;
+import com.biegan.game.View.sprites.*;
+
+public class GameController {
+
+    public enum GAME_STATE
+    {
+        START,
+        NOT_STARTED,
+        STARTED,
+        OVER
+    };
+
+    private GameModel model;
+    private GAME_STATE state;
+    private BieganBrickRaceGame game;
+    private Screen currScreen;
+    private GameScreen gameScreen;
+    private GameInput gameInput;
+    private OverInput overInput;
+    private int score;
+    private int lastSpeedIncreaseScore = 0;
+    private boolean isGameStarted;
+    private float gameSpeed;
+
+    private RoadStripes roadStripes;
+    private Bushes bushes;
+    private EnemyCarManager enemyCarManager;
+    private Player player;
+    private Array<Explosion> explosions;
+    private Hud hud;
+    private PositionsX positionsX;
+
+    private final float worldWidth = 800; // Width of game world

```

Comment:
controller nie ma żadnego interesu tego wiedzieć, to wie albo model, jeśli liczy jakieś kolizje or whatever, albo view, żeby wyświetlać coś w konkretnych przedziałach worldWidth (albo oba), ale na pewno nie controller

Reply:
tego, ani żadnych wyżej. Controller ma tylko znać stan gry, żeby móc notyfikować model/view i wybrać odpowiedni inputprocessor. A poza tym controller zajmuje się tylko przekazywaniem danych z jednego miejsca w drugie. Czyli model liczy wszystko istotne dla logiki gry: położenie gracza i przeciwników, prędkość, punkty i sprawdza po update'cie położeń, czy nie doszło do kolizji. View prosi controller o te dane w każdej pętli, żeby odpowiednio wyświetlić gracza i przeciwników (wyświetlaniem reszty view zajmuje się sam), i model notyfikuje controller jak jest kolizja, bo to game over, inputprocessor powinien być częścią view i view powinno notyfikować controller jak jest pauza. Chyba musimy się spotkać jeszcze raz to obgadać